### DISCUSSION TITLE
Can iframe tag be used?

### DISCUSSION THREAD

**Discussion by ApprenticeGC** [outline/outline#2286] [Self Hosting]
Just want to know if iframe tag could be used inside the doc?

**Comment by tommoor** [comment]
Not directly, we have embeds to support individual types of integrations at the moment so it depends on what you're trying to embed

**Reply by ApprenticeGC** [reply]
I am using outline for game design doc. Occasionally, I need to embed Sketchfab using iframe to show referenced model or animation. Besides, I embed Unity asset store product info using iframe frequently.

The use of iframe has many possibilities. Hope outline can consider adding generic iframe support.

**Comment by chrisshaw** [comment]
Is this the place to track requests for general iframe integration? It's a question, not a feature request, but I didn't see a feature request.

:+1: for generic iframe support.

**Reply by EduNetEurope** [reply]
Hi @tommoor,

according to the [Outline Help](https://docs.getoutline.com/s/guide/doc/embeds-bqUBtgpqR0), it is possible to use iframes:
> Outline supports a variety of interactive embeds for services such as video, spreadsheets, forms, and more. Embeds can be inserted by pasting a compatible url, iframe code, or using the / menu and searching for the service.

Where do we need to past the iframe code?
How can we embed an iframe?

[#1815](https://github.com/outline/outline/pull/1815)
seems to have been abandoned?

**Reply by tommoor** [reply]
There is support for pasting iframes from the supported services, in the same way as a regular link. Wider iframe support is not landed yet, this is a technical limitation blocked by #3000.

**Comment by johnoscott** [comment]
This sounds like an obvious addition to Outline that will unlock just about any integration that is iframe based. @tommoor what do you think ?

**Reply by danielvianna** [reply]
I have a use case for generic Iframe. I'm testing whether outline can be used for documentation of design systems, instead of using storybook which only devs can use

**Comment by adrianmarkperea** [comment]
Hello! Is there an issue we can track to see the progress of generic iframe integration?

**Comment by ReessKennedy** [comment]
+1 for this ... seems like this would cover many types of embeds. I am particularly interested in embeds for the NoCode Airtable alternatives Baserow and Nocodb .

**Reply by ReessKennedy** [reply]
ONE BIGGER ARGUMENT FOR THIS: I sometimes will create visualizations using an API and then I'll just output those visualizations to a public HTML page ... and I would really like to be able to embed these visualizations inside outline. If I screenshot an image they won't stay updated but if I can control all my data in one place and know that the visualizations for the data can be linked in our team docs and remain updated because of an embed, that would be really helpful.

**Reply by deletosh** [reply]
This is indeed a big use-case — smaller, alternative services that are gaining traction.

In my case, I prefer BerryCast to Loom. There's support for loom, and have generic iframe will make BerryCast work.

**Reply by tommoor** [reply]
There is already support for BerryCast fwiw

**Reply by dolsem** [reply]
@tommoor I would also love to have NocoDB support - it's a neat self-hosted solution that can turn any external SQL database into an AirTable-like spreadsheet. I'd like to use it as a task tracker inside Outline. I'm already almost all set with a PR for this, just wanted to confirm there is interest per contribution guidelines. I've also gotten a few PRs merged into NocoDB to make the integration better and more powerful. And I've just submitted this PR - https://github.com/outline/outline/pull/4134 that helps my use case as well.

**Comment by fuzyll** [comment]
I'd like to throw my hat in the ring for this as well. I actually have a Google Calendar embed that Outline is currently rejecting. I can create an issue for this (since I presume it should work), but I also think it's silly to spend developer time/resources tracking down every little issue with remote services. I would be perfectly happy, as a user, with just falling back to more generic `iframe` support to solve my specific problem.

**Comment by snarkyalyx** [comment]
@tommoor The fact that I have to use the Self Hosted draw.io deployment field to let us use self-hosted grist embeds is really bothering me. Can you please just make a table where admins can just add links that allow you to do iframe embeds?

**Comment by Anomen** [comment]
I'll get a stab at that. I need it too. I'll try to open a PR soon.

**Reply by Anomen** [reply]
https://github.com/outline/outline/pull/5137

**Reply by snarkyalyx** [reply]
Something people have been asking for years took this man less than 40 lines of code and under an hour of work lmao, good effort! Hope it makes it.

**Reply by Anomen** [reply]
I really don't deserve the merit. The code base is quite clear, so congrats to the main contributors 🙌.
I also happen to have a good knowledge of React and Nodejs, so it was right in my alley. 
Happy to help!

**Reply by routerino** [reply]
> I really don't deserve the merit. The code base is quite clear, so congrats to the main contributors 🙌. I also happen to have a good knowledge of React and Nodejs, so it was right in my alley. Happy to help!

Question, can we use this same method to enable https://github.com/outline/outline/issues/2831? That's been on my radar for a while, to the point where I was considering learning react just to implement a proof of concept for it.
